src/tests/functional/plugin/gna/shared_tests_instances/subgraph_tests/basic_lstm_sw_fp32.cpp

```diff
@@ -0,0 +1,134 @@
+        outFormShapes1 = { batch_size, reshape1_shape[2] };
+        auto constantX = std::make_shared<ngraph::opset1::Constant>(ngraph::element::i64, ngraph::Shape{ 2 }, outFormShapes1);
+        auto lstm1 = std::make_shared<ngraph::opset4::LSTMCell>(std::make_shared<ngraph::opset1::Reshape>(X, constantX, false),
+            H_t, C_t,
+            weightsNode, reccurrenceWeightsNode, hiddenSize);
+
+        auto H_o = lstm1->output(0);
+        auto C_o = lstm1->output(1);
+
+        //TensorIterator [1, num_cells, thirdDimOut] [1, 118], [1, 118] -> [1, 118]
+        auto body = std::make_shared<ngraph::Function>(
+            ngraph::OutputVector{ H_o, C_o }, ngraph::ParameterVector{ X, H_t, C_t });
+
+        auto tensor_iterator = std::make_shared<ngraph::opset1::TensorIterator>();
+        tensor_iterator->set_body(body);
+
+        //input tensor shape: [1, num_cells, thirdDimOut] chunk shape: [1, 1, thirdDimOut]
+        tensor_iterator->set_sliced_input(X, reshape1, 0, 1, 1, -1, 1);
+        tensor_iterator->set_merged_input(H_t, H_init, H_o);
+        tensor_iterator->set_merged_input(C_t, C_init, C_o);
+
+        auto out0 = tensor_iterator->get_iter_value(H_o, -1);
+
+        const size_t output_size = 12;
+        auto fc1 = ngraph::builder::makeFullyConnected(out0, ngPrc, output_size, true, { hiddenSize, output_size }, { 0.05f }, { 0.1f });
+
+        ngraph::ResultVector results{ std::make_shared<ngraph::opset1::Result>(fc1) };
+        return std::make_shared<ngraph::Function>(results, params, "Basic_LSTM_S");
+    }
+
+protected:
+    InferenceEngine::Blob::Ptr GenerateInput(const InferenceEngine::InputInfo &info) const override {
+        return FuncTestUtils::createAndFillBlob(info.getTensorDesc(), 0.05f, 0.f);
+    }
+
+    void SetUp() override {
+        threshold = 0.1f;
+
+        InferenceEngine::Precision netPrecision;
+        std::pair<size_t, size_t> size_params;
+        size_t num_cells;
+        bool decompose;
+        std::tie(netPrecision, targetDevice, configuration, size_params, num_cells, decompose) = this->GetParam();
+        third_dim = size_params.first;
+        hidden_size = size_params.second;
+        outPrc = InferenceEngine::Precision::FP32;
+
+        function = GetNetwork_sw_fp32(size_params.first, size_params.second, num_cells, netPrecision, &hidden_memory_init, &cell_memory_init);
+        if (decompose) {
+            ngraph::pass::Manager manager;
+            manager.register_pass<ngraph::pass::LSTMCellDecomposition>();
+            manager.run_passes(function);
+        }
+    }
+};
+
+TEST_P(Basic_LSTM_SW_FP32, CompareWithRefImpl) {
+    Run();
+};
+
+const std::vector<InferenceEngine::Precision> netPrecisions = {
+        InferenceEngine::Precision::FP32,
+        InferenceEngine::Precision::FP16
+};
+
+std::map<std::string, std::string> config_no_sf = {
+    {"GNA_COMPACT_MODE", "NO"},
+    {"GNA_DEVICE_MODE", "GNA_SW_FP32"}
```

Comment:
We need to test the both GNA_SW_EXACT and GNA_SW_FP32 modes every time if it is possible